src/register.php

```diff
@@ -27,40 +27,27 @@
 				exit("This email is already registered");
 			}
 
-			// collecting information for validation
-			$time = time(); // in unix format
-			$validation = md5($email . $time . rand());
-			// ignore IP for now due to ip2long() not being able to handle IPv6 addresses
-			// TODO: remove this column from the database
-			$IP = 0;
-
-			$sth = $dice->dbconn->prepare("SELECT email FROM pending_validations WHERE email=?");
-			$sth->bind_param('s', $email);
-			$sth->execute() or trigger_error($mysqli->error);
+			// generating a random token for registration validation
+			// the DB table allows entries up to 32 chars
+			$validation = base64_encode(random_bytes(24));
 
-			$isNewRegistration = $sth->num_rows === 0;
-			$sth->close();
-			// insert or update pending validation
-			if ($isNewRegistration) {
-				$sql = "INSERT INTO pending_validations (email, validation_key, time_stamp, IP) VALUES (?, ?, FROM_UNIXTIME(?), ?)";
-				$sth = $dice->dbconn->prepare($sql);
-				$sth->bind_param('ssss', $email, $validation, $time, $IP);
-				$sth->execute();
-			} else {
-				$sql = "UPDATE pending_validations SET validation_key=?, time_stamp=FROM_UNIXTIME(?), IP=? WHERE email=?";
-				$sth = $dice->dbconn->prepare($sql);
-				$sth->bind_param('ssss', $validation, $time, $IP, $email);
-				$sth->execute();
-			}
+			// ignore IP for now due to ip2long() not being able to handle IPv6 addresses
+			// TODO: remove this column from the database, remove time_stamp as well
+			// because it is ignored as well if the timestamp doesn't provide much value
+			$sql = "REPLACE INTO pending_validations (email, validation_key, IP) VALUES (?, ?, 0)";
+			$sth = $dice->dbconn->prepare($sql);
+			$sth->bind_param('ss', $email, $validation);
+			$sth->execute() or trigger_error($dice->dbconn->error);
 			$sth->close();
+
 			// sending email
 			$to = $email;
 			$email_enc = urlencode($email);
 			$subj = "Registration for MARTI dice server";
 			$from = "marti@tripleawarclub.org";
 			$ehead= "From: MARTI<$from>\n";
 			$ehead .= "List-Unsubscribe:<$dice->domain/unsubscribe.php?email=$email_enc>\n";
-			$message = "To validate your email click this link: $dice->domain/validate.php?email=$email_enc&val=$validation";
+			$message = "To validate your email click this link: $dice->domain/validate.php?email=$email_enc&val=" . urlencode($validation);
```

Comment:
This change is needed, because base64 uses some characters that would mess with the URL scheme/ are not allowed in this context